third_party/chromium/media/base/decoder_buffer.h

```diff
@@ -235,6 +235,9 @@ class MEDIA_EXPORT DecoderBuffer
     is_key_frame_ = is_key_frame;
   }
 
+  // Returns total memory used for bookkeeping and buffered data.
+  virtual size_t total_memory_usage() const;
```

Comment:
1. Consider making this non-virtual if the `preroll_buffer_` in StreamParserBuffer is rarely used.  While this makes the return value of the function less accurate, it avoids introducing the first virtual function (beyond the dtor) here.  If we move ahead with this, we should refine the comment accordingly.
2. We should also call out that this is meant to be used for buffer eviction.
3. Given the comment, `total_memory_used` might be a better function name, or maybe just `memory_used`.